src/main/kotlin/three/consulting/epoc/entity/Customer.kt

```diff
@@ -10,5 +11,12 @@ data class Customer(
     @field:Column(nullable = false) var created: LocalDateTime = LocalDateTime.now(),
     @field:Column(nullable = false) var updated: LocalDateTime = LocalDateTime.now(),
     @field:Column(nullable = false) var enabled: Boolean = true,
-    @field:Id @field:GeneratedValue(strategy = GenerationType.IDENTITY) val id: Long = -1
-)
+    @field:Id @field:GeneratedValue(strategy = GenerationType.IDENTITY) val id: Long? = null
+) {
+    constructor(customerDTO: CustomerDTO) : this (
+        id = customerDTO.id,
+        name = customerDTO.name,
+        description = customerDTO.description,
+        enabled = customerDTO.enabled ?: true
+    )
```

Comment:
You should also include the created-timestamp in this. Without it it might get overridden with the current timestamp every time you save an entity, even in future updates.